tests/unit/test_utils_pyc_compile.py

```diff
@@ -0,0 +1,172 @@
+from contextlib import contextmanager
+from functools import partial
+from pathlib import Path
+from typing import Iterator, Optional, Type
+from unittest.mock import Mock, patch
+
+import pytest
+from pytest import param  # noqa: PT013
+
+from pip._internal.utils import pyc_compile
+from pip._internal.utils.pyc_compile import (
+    BytecodeCompiler,
+    ParallelCompiler,
+    SerialCompiler,
+    _compile_single,
+    create_bytecode_compiler,
+)
+
+try:
+    import concurrent.futures
+    import multiprocessing
+except (OSError, NotImplementedError, ImportError):
+    parallel_supported = False
+else:
+    parallel_supported = True
+
+needs_parallel_compiler = pytest.mark.skipif(
+    not parallel_supported, reason="ParallelCompiler is unavailable"
+)
```

Comment:
This could be me being simply paranoid, but I wanted the tests to continue to work even on systems where multiprocessing is broken, hence this marker.